bech32/bech32.go

```diff
@@ -229,17 +261,17 @@ func DecodeNoLimit(bech string) (string, []byte, error) {
 	}
 
 	// We exclude the last 6 bytes, which is the checksum.
-	return hrp, decoded[:len(decoded)-6], nil
+	return hrp, values, nil
 }
 
 // Decode decodes a bech32 encoded string, returning the human-readable part and
 // the data part excluding the checksum.
 //
 // Note that the returned data is 5-bit (base32) encoded and the human-readable
 // part will be lowercase.
-func Decode(bech string) (string, []byte, error) {
-	// The maximum allowed length for a bech32 string is 90.
-	if len(bech) > 90 {
+func Decode(bech string, limit int) (string, []byte, error) {
```

Comment:
This change is the same as the enigma change which the SDK already uses IIRC, so not API breaking for the cosmos SDK